tests/reftests/gen.ml

```diff
@@ -95,15 +95,31 @@ let () =
   in
   let process archive_hashes filename =
     let base_name = OpamStd.String.remove_suffix ~suffix:".test" filename in
-    let condition = match Filename.extension base_name with
-      | "" -> ""
-      | os ->
-        Printf.sprintf "\n (enabled_if (= %%{os_type} %S))"
-          String.(capitalize_ascii (sub os 1 (length os - 1)))
-    in
     if base_name = filename then archive_hashes else
+      let archive_hash, tags =
+        match first_line_tags ~path:filename with
+        | [] -> "", []
```

Comment:
`String.split_on_char` never returns a empty list. Instead it is `[""]` (see https://ocaml.org/manual/5.2/api/String.html#extract)

Reply:
I can add a comment for this, or fix the function, but `[]` and `[""]` end up with the exact same result anyways.

Reply:
i think replacing it with `assert false` would be a bit nicer